internal/isoutil/isoutil.go

```diff
@@ -228,3 +236,73 @@ func (h *installerHandler) VolumeIdentifier() (string, error) {
 
 	return strings.TrimSpace(string(volumeId)), nil
 }
+
+func GetFileLocation(filePath, isoPath string) (int64, error) {
+	d, err := diskfs.OpenWithMode(isoPath, diskfs.ReadOnly)
+	if err != nil {
+		return 0, err
+	}
+
+	fs, err := d.GetFilesystem(0)
+	if err != nil {
+		return 0, err
+	}
+
+	isoFile, err := fs.OpenFile(filePath, os.O_RDONLY)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Failed to open file %s", filePath)
+	}
+
+	myFile := isoFile.(*iso9660.File)
+	defaultSectorSize := int64(2 * 1024)
+	offset := int64(myFile.Location()) * defaultSectorSize
+	return offset, nil
+}
+
+// Writing the offsets of initrd images in the end of system area (first 32KB).
+// As the ISO template is generated by us, we know that this area should be empty.
+func WriteHeader(ignitionOffsetInfo, ramDiskOffsetInfo *OffsetInfo, isoPath string) error {
+	iso, err := os.OpenFile(isoPath, os.O_WRONLY, 0o664)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	// Starting to write from the end of the system area in order to easily support
+	// additional offsets (and as done in coreos-assembler/src/cmd-buildextend-live)
+	headerEndOffset := int64(32768)
+
+	// Write ignition config
+	headerEndOffset, err = writeOffsetInfo(headerEndOffset, ignitionOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	// Write ram disk
+	_, err = writeOffsetInfo(headerEndOffset, ramDiskOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeOffsetInfo(headerOffset int64, offsetInfo *OffsetInfo, iso *os.File) (int64, error) {
```

Comment:
Same here, this is only needed in the editor.

Reply:
done